thumbor/engines/pil.py

```diff
@@ -263,6 +265,14 @@ def read(self, extension=None, quality=None):  # NOQA
 
         results = img_buffer.getvalue()
         img_buffer.close()
+
+        if self.context.config.SERVE_ORIGINAL_IF_SMALLER and hasattr(self.context.request.filters, '__len__') and len(self.context.request.filters) == 0:
```

Comment:
Does not seem like a good idea to do it in one engine. Probably would be better to make it in handler somewhere near place where it tries to fit into bytes.

Reply:
Good point, thanks. I'll see if I can find a better spot to put that!